### DISCUSSION TITLE
Set non-uniform initial condition of alphaPorosity

### DISCUSSION THREAD

**Discussion by chairmanmao256** [mdolab/dafoam#201] [General]
Hello! I'm here again!
**This time, I'm here to ask how to set a non-uniform initial condition of alphaPorosity.** I've already taken a few steps to achieve this goal. But I met some problems. 
I'd like to set alphaPorosity to 1e4 in all cells within the box defined by myself. I managed to get all the cell-index (global index)  of the cells that locate in the box. The index-list is stored in a file called: 'cells.txt', which is as follows:
```
Add index: 5144
Add index: 5145
Add index: 5146
Add index: 5147
Add index: 5148
...
```
Then I use this index-list to set the initial value of alphaPorosity in `runScript.py`:
```
alphaPorosity0 = np.zeros(nCells, dtype="d")
cells = open("cells.txt",'r')
cellsData = cells.readlines()
cells.close()

for line in cellsData:
    index = line.split()
    alphaPorosity0[int(index[2])] = 1e4
```
Everything is OK when I run DAFoam using only one core. (i.e. The alphaPorosity is set to 1e4 in the box specified by myself.) However, if I run DAFoam in parallel mode, the real initial field of alphaPorosity is by no means similar to the initial field I want to set.(i.e. The alphaPosority distributes randomly in the entire field instead of accumulating in the box specified by myself.)

I believe that the function `setFieldValue4GlobalCellI` has something to do with this problem. [As I understanded, the function just change the `globalIndex` (read from the alphaPorosity array `alphaPorosity0` specified in `runScript.py`) to `localIndex`.] But I can't tell where the problem is.

Thank you!

**Comment by friedenhe** [comment]
You cant use the same indices for serial and parallel cases. Check this post: https://github.com/mdolab/dafoam/discussions/102

**Reply by chairmanmao256** [reply]
Thanks a lot!
